plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/tabbar/TabBarTest.java

```diff
@@ -534,4 +537,58 @@ private void checkDiagramTabbarButtons(boolean activeExtensions) {
         }
         checkButtonNotPresent(TABBAR_EXTENSION_ON_DIAGRAM_ELEMENT);
     }
+
+    public void testToolBarWrap() throws Exception {
+        SWTBotUtils.waitAllUiEvents();
+        SWTBotShell window = bot.shell();
+        syncExec(() -> {
+            // resize the window so there's not enough space for all the tools in the toolbar (in width)
+            window.widget.setSize(640, 480);
+        });
+
+        // diagram tools
+        // get all tool of toolbar
+        var toolbarDiagramItems = new ArrayList<SWTBotToolbarButton>();
+        for (int i = 0; i < DIAGRAM_TOOLBARBUTTONS_TOOLTIPS.length; ++i) {
+            toolbarDiagramItems.add(editor.bot().toolbarButton(i));
+        }
+        for (int i = 0; i < DIAGRAM_TOOLBARDROPDOWNBUTTONS_TOOLTIPS.length; ++i) {
+            toolbarDiagramItems.add(editor.bot().toolbarDropDownButton(i));
+        }
+
+        // we check that no tools are outside the visible part of the toolbar
+        for (SWTBotToolbarButton item : toolbarDiagramItems) {
+            boolean itemIsOutside = syncExec(() -> {
+                int parentWidth = item.widget.getParent().getSize().x;
+                int rightPos = item.widget.getBounds().x + item.widget.getBounds().width;
+                return rightPos <= parentWidth;
+            });
+            assertTrue("ToolButton `" + item.getToolTipText() + "` is outside the visible part of the toolbar.", itemIsOutside);
```

Comment:
The test fail on my environment: junit.framework.AssertionFailedError: ToolButton `Copy the format of the whole diagram` is outside the visible part of the toolbar.